### ISSUE TITLE
Handle recursive structures in instance variables

### ISSUE THREAD

**Issue by prehnRA** [revelrylabs/execjs-rails#10]
Sometimes controller instance variables recursively reference themselves. This causes the renderer to fail in the default config when we attempt to create the props JSON.

We need to figure out how to address it. Some ideas:
- Use a JSON serializer that gracefully handles recursive structures.
- Set a max depth for the props structure.
- Make it easier to blacklist assigns from being in the props, and set some intelligent default blacklist (Devise is a big culprit here, plus we probably want to exclude devise's structures by default anyway).

**Comment by jwietelmann** [comment]
I'm actually starting to feel that 1.0 should completely punt on serialization of the view. The default we have now seems antithetical to the gem's mission, which is to be utility upon which opinions can be built, while having as few of its own as possible.

**Comment by prehnRA** [comment]
Which would basically mean requiring users of the gem to provide a configuration for the view to options lambda. If we don't want to build that lambda into the gem directly, maybe we could add a generator that sets up an initializer with all of the tweakable options?

**Comment by jwietelmann** [comment]
That's vaguely the idea. Default `view_to_function_options` lambda still takes `view` and `options` args but returns nil. The generator idea would be nice I guess, but just documenting the available defaults and overrides in the README would probably be sufficient (and more necessary). I don't see a generator being used often or at all, since this gem is targeted toward building other JS-based view renderers. If someone wants to routinely install execjs-rails into Rails apps instead of a wrapper gem, that person is probably doing something pretty terrible.

**Comment by prehnRA** [comment]
Yes, but, the default behavior is probably something that a lot of frameworks would use. It is so simple: make the same variables which are available to an ERB available to a javascript function. I could see almost any wrapper using that same logic.
